sdk/storage/Azure.Storage.Common/src/UploadTransactionalHashingOptions.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Storage
+{
+    /// <summary>
+    /// Hashing options for validating upload content.
+    /// </summary>
+    public class UploadTransactionalHashingOptions
+    {
+        /// <summary>
+        /// Hashing algorithm to use for the transaction.
+        /// </summary>
+        public TransactionalHashAlgorithm Algorithm { get; set; }
+
+        /// <summary>
+        /// Optional. For an upload API that ends up using a single REST call,
+        /// a precalculated hash for the upload contents can be provided to skip
+        /// recalculating. This hash will be ignored if the upload splits into
+        /// multiple REST calls and new ones will be calculated.
```

Comment:
are we going to ignore or throw? if it gets ignored it's a trap.

Reply:
We're supposed to throw. Likely this is an old doc that needs updating but I will check the code as well.